packages/-ember-data/addon/setup-container.js

```diff
@@ -1,43 +1,36 @@
 import { DebugAdapter } from './-private';
 import Store from '@ember-data/store';
-import JSONAPISerializer from '@ember-data/serializer/json-api';
-import JSONSerializer from '@ember-data/serializer/json';
-import RESTSerializer from '@ember-data/serializer/rest';
 import JSONAPIAdapter from '@ember-data/adapter/json-api';
 import RESTAdapter from '@ember-data/adapter/rest';
 
-import { BooleanTransform, DateTransform, NumberTransform, StringTransform } from '@ember-data/serializer/-private';
-
-function has(applicationOrRegistry, fullName) {
-  if (applicationOrRegistry.has) {
-    // < 2.1.0
-    return applicationOrRegistry.has(fullName);
-  } else {
-    // 2.1.0+
-    return applicationOrRegistry.hasRegistration(fullName);
+function hasRegistration(application, registrationName) {
+  // fallback our ember-data tests necessary
+  // until we kill-off createStore/setupStore
+  // or @ember/test-helpers kills off it's
+  // legacy support that calls our initializer with registry
+  // instead of application
+  if (!application.hasRegistration) {
```

Comment:
No need to do boolean coercion here...

```suggestion
  if (typeof application.hasRegistration !== 'function') {
```